### DISCUSSION TITLE
Controlling the python version without live prompts?

### DISCUSSION THREAD

**Discussion by tlc** [astral-sh/rye#597] [Q&A]
I've inherited a rye based Python project.  I don't understand how to control the version of python.   ry-install.sh is called unattended (with RYE_INSTALL_OPTION=--yes), so I can't answer the prompt `Which version of Python should be used as default toolchain?`

The .python-version file has `3.10.13`.  The pyproject.toml file has
`requires-python = "== 3.10"
`

Today I noticed that the latest rye (0.21.0) installs python 3.12, which causes problems for us with a deprecated module.
Earlier ryes had been installing 3.11, but we hadn't noticed.

I couldn't figure out how to force a python version, so I forced an older rye with RYE_INSTALL, so we'd get 3.11.

How am I supposed to control the version of python?

**Comment by mitsuhiko** [comment]
Rye itself installs 3.12 for it's internal uses. For the project it should be picking up the version mentioned in the `.python-version` file. If that is not happening please paste the output of `rye sync` after wiping `.venv`.
